stellargraph/layer/node2vec.py

```diff
@@ -0,0 +1,209 @@
+        Construct node representations from node ids through a look-up table.
+
+        Args:
+            xin (Keras Tensor): Batch input node ids.
+            embedding (str): "input" for input_embedding matrix, "output" for output_embedding
+
+        Returns:
+            Output tensor.
+        """
+
+        if embedding == "input":
+            h_layer = self.input_embedding(xin)
+        elif embedding == "output":
+            h_layer = self.output_embedding(xin)
+        else:
+            raise ValueError(
+                'wrong embedding argument is supplied: {}, should be "input" or "output"'.format(
+                    embedding
+                )
+            )
+
+        h_layer = Reshape((self.emb_size,))(h_layer)
+
+        return h_layer
+
+    def _node_model(self, embedding="input"):
+        """
+        Builds a Node2Vec model for node prediction.
+
+        Args:
+            embedding (str): "input" for input_embedding, "output" for output_embedding
+
+        Returns:
+            tuple: (x_inp, x_out) where ``x_inp`` is a Keras input tensor
+            for the Node2Vec model and ``x_out`` is the Keras tensor
+            for the Node2Vec model output.
+
+        """
+        # Create tensor inputs
+        x_inp = Input(shape=(1,))
+
+        # Output from Node2Vec model
+        x_out = self(x_inp, embedding)
+
+        return x_inp, x_out
+
+    def _link_model(self):
+        """
+        Builds a Node2Vec model for link or node pair prediction.
+
+        Returns:
+            tuple: (x_inp, x_out) where ``x_inp`` is a list of Keras input tensors for (src, dst) nodes in the node pairs
+            and ``x_out`` is a list of output tensors for (src, dst) nodes in the node pairs.
+
+        """
+        # Expose input and output sockets of the model, for source node:
+        x_inp_src, x_out_src = self._node_model("input")
+        x_inp_dst, x_out_dst = self._node_model("output")
+
+        x_inp = [x_inp_src, x_inp_dst]
+        x_out = [x_out_src, x_out_dst]
+        return x_inp, x_out
+
+    def in_out_tensors(self, multiplicity=None):
+        """
+        Builds a Node2Vec model for node or link/node pair prediction, depending on the generator used to construct
+        the model (whether it is a node or link/node pair generator).
+
+        Returns:
+            tuple: (x_inp, x_out), where ``x_inp`` contains Keras input tensor(s)
+            for the specified Node2Vec model (either node or link/node pair model) and ``x_out`` contains
+            model output tensor(s) of shape (batch_size, self.emb_size)
+
+        """
+        if multiplicity is None:
+            multiplicity = self.multiplicity
+        if self.multiplicity == 1:
+            return self._node_model()
+        elif self.multiplicity == 2:
+            return self._link_model()
+        else:
+            raise RuntimeError(
+                "Currently only multiplicities of 1 and 2 are supported. Consider using node_model or "
```

Comment:
This error message should be updated to remove the mention of `node_model` and `link_model`

Reply:
I will make the update.